src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,287 @@
+                                url=url,
+                                params=params,
+                                headers=headers,
+                                json=json,
+                            )
+                            break
+                    except asyncio.TimeoutError:
+                        print(
+                            f"Client request timeout occurred while connecting to the API. Retrying in {self.polling_interval} seconds..."
+                        )
+                        self.polling_step += 1
+                        await asyncio.sleep(self.polling_interval)
+                    except (ClientError, socket.gaierror) as exception:
+                        raise ConnectionError(
+                            "Error occurred while communicating with the API."
+                        ) from exception
+        except asyncio.TimeoutError as exception:
+            raise ConnectionError(
+                "Client polling timeout while connection to the API."
+            ) from exception
+
+        check_content_type(response)
+
+        return cast(dict[str, Any], await response.json()), response.status
+
+    async def request_once(
+        self,
+        method: str,
+        url: str,
+        params: dict[str, Any] | None = None,
+        headers: dict | None = None,
+        json: dict | None = None,
+    ):
+        """Sends a single request to FlexMeasures and checks the response"""
+        response = await self.session.request(
+            method=method,
+            url=url,
+            params=params,
+            headers=headers,
+            json=json,
+            ssl=self.ssl,
+        )
+        await check_response(
+            self,
+            response,
+        )
+        print(response.headers)
+        return response
+
+    def client_should_retry(self, exception, response) -> bool:
+        """Determines if the client should retry because the job is not yet finished"""
+        payload = response.json()
+        return getattr(exception, "status") == 400 and (
+            "Scheduling job waiting" in payload.get("message", "")
+            or "Scheduling job in progress" in payload.get("message", "")
+        )
+
+    def start_session(self):
+        """If there is no session, start one"""
+        if self.session is None:
+            self.session = ClientSession()
+
+    async def get_headers(self, include_auth: bool) -> dict:
+        """If the request needs to be authenticated check if there is a access_token or request one. Then create the headers dict"""
+        headers = CONTENT_TYPE_HEADERS
+        if include_auth:
+            if self.access_token is None:
+                await self.get_access_token()
+            headers |= {"Authorization": self.access_token}
+        print(headers)
+        return headers
+
+    def build_url(self, uri: str, path: str =path):
+        if "://" in self.host:
+            self.host = self.host.split("://")[1]
```

Comment:
I'm not sure if we want the scheme to be updated after the initialization of the client. But maybe this check for :// should not be there but some validator should be placed on the host variable. (I do not know how this would be done nicely)

Reply:
currently removed this function and created an issue for validation on `__init__` / `__post_init__`: https://github.com/FlexMeasures/flexmeasures-client/issues/11